prisma/schema.prisma

```diff
@@ -10,19 +10,71 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
+enum StationType {
+  Fixed
+  Mobile
+}
+
 model Station {
-  id   String @id @default(auto()) @map("_id") @db.ObjectId
-  name String
-  type String
-  sensors Sensor[]
-  deletedAt DateTime?
+  id            String   @id @default(auto()) @map("_id") @db.ObjectId
+  name          String
+  type          StationType
+  latitude      String
+  longitude     String
+  description   String
+  image_url     String
+  sensors       Sensor[]   // Define a relation to Sensor
+}
+
+enum SensorType {
+  LittObs
+  SensOcean
 }
 
 model Sensor {
-  id   String @id @default(auto()) @map("_id") @db.ObjectId
-  identifier String @unique
-  stationId String? @db.ObjectId
-  station Station? @relation(fields: [stationId], references: [id])
+  id            String   @id @default(auto()) @map("_id") @db.ObjectId
+  identifier    String   @unique
+  type          SensorType
+  nbr_measures  Int
+  stationId     String?  @db.ObjectId // Add a reference to the Station model
+  station       Station? @relation(fields: [stationId], references: [id])
+  records       Record[]    // Add a one-to-many relationship to Record
+  files         File[]      // Add a one-to-many relationship to File
+}
+
+model Record {
+  id                  String   @id @default(auto()) @map("_id") @db.ObjectId
+  latitude            String
+  longitude           String
+  recorded_at         DateTime @default(now())
+  battery_voltage     Float
+  battery_percentage  Float
+  pression_ext        Float
+  temp_ext            Float
+  temp_int            Float
+  temp_sea            Float[]
+  temp_sea_mean       Float
+  ec_sea              Float[]
+  ec_sea_mean         Float 
+  sensorId            String?  @db.ObjectId  // Reference to Sensor model
+  sensor              Sensor?  @relation(fields: [sensorId], references: [id])
+}
+
+model File {
+  id            String   @id @default(auto()) @map("_id") @db.ObjectId
+  name          String?
+  update_date   DateTime
+  status        String?
+  file_url      String?
+  sensorId      String?  @db.ObjectId  // Reference to Sensor model
+  sensor        Sensor?  @relation(fields: [sensorId], references: [id])
+}
```

Comment:
same here for properties to track user modification

Reply:
@cballevre, since we are tracking user modifications in multiple places, its not better to create an audit table? This would have all the info of created_at, created_by, updated_at, updated_by and more like: edited from this to this, deleted_at, deleted_by... 

With this table, would be also possible to create a page in the future just for managing the auditions (a place were managers can track modifications in the system).

Reply:
In a relational database this would have suited me fine. But with document databases, I prefer denormalisation. Maybe can we use the composite types from Prisma ([doc](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-composite-types)) so we don't have to repeat all over again the same properties
```
model Station {
  id     String  @id @default(auto()) @map("_id") @db.ObjectId
  name         String?
  status        String?
  audit.         Audit
}

type Audit {
  created_at
  created_by
  updated_at
  updated_by
  deleted_at
  deleted_by
}
```